configure.ac

```diff
@@ -570,6 +570,38 @@ if test "x$enable_backtrace" = xunwind-ptrace; then
    ]])], [AC_DEFINE([HAVE_LIBUNWIND_ELF_FILENAME], [1], [Define if libunwind has unw_get_elf_filename])], [])
 fi
 
+AC_ARG_ENABLE([demangling],
+              [AS_HELP_STRING([--enable-demangling],
+                              [enable demangling support for backtraces @<:@default=check@:>@])],
+              [],
+              [enable_demangling=check])
+case "$enable_demangling" in
```

Comment:
My attempt to rewrite the code:
```shell

have_libiberty=no
case $enable_demangling in
   check|yes|libiberty)
      AC_CHECK_LIB([iberty], [cplus_demangle], [have_libiberty=yes], [
         if test "x$enable_demangling" = xlibiberty; then
            AC_MSG_ERROR([--enable-demangling specified but libiberty not found])
         fi
      ])
      if test "x$have_libiberty" = xyes; then
         AC_CHECK_HEADERS([demangle.h], [], [
dnl FIXME: do i need to restore $CPPFLAGS after this?
            CPPFLAGS="$CPPFLAGS -I/usr/include/libiberty"
            AC_CHECK_HEADERS([demangle.h], [], [
               have_libiberty=no
               if test "x$enable_demangling" = xlibiberty; then
                  AC_MSG_ERROR([--enable-demangling specified but <demangle.h> not found])
               fi
            ])
         ])
      fi
      if test "x$have_libiberty" = xyes; then
         enable_demangling=libiberty
      fi
      ;;
esac

dnl have_libXXX=no
dnl case $enable_demangling in
dnl    check|yes|libXXX)
dnl       AC_CHECK_LIB([XXX], [funcname], [have_libXXX=yes], [
dnl          if test "x$enable_demangling" = xlibXXX; then
dnl             AC_MSG_ERROR([--enable-demangling specified but libXXX not found])
dnl          fi
dnl       ])
dnl       if test "x$have_libXXX" = xyes; then
dnl          AC_CHECK_HEADERS([XXX.h], [], [
dnl             have_libXXX=no
dnl             if test "x$enable_demangling" = xlibXXX; then
dnl                AC_MSG_ERROR([--enable-demangling specified but <XXX.h> not found])
dnl             fi
dnl          ])
dnl       fi
dnl       if test "x$have_libXXX" = xyes; then
dnl          enable_demangling=libXXX
dnl       fi
dnl       ;;
dnl esac

case $enable_demangling in
   check)
      enable_demangling=no
      ;;
   yes)
      AC_MSG_ERROR([cannot find any library for the backend supported by --enable-demangling])
      ;;
   *)
      AC_MSG_ERROR([bad value '$enable_demangling' for --enable-demangling])
      ;;
esac
```

Reply:
Thank you very much @Explorer09.
Your code inspired me to improve what I wrote.

Reply:
Oops. I made a mistake in the example code given above. It should be written like this instead:

```m4
case $enable_demangling in
   libiberty)
      AC_DEFINE([HAVE_DEMANGLING], [1], [Define to 1 if the demangling is supported])
      ;;
   check)
      enable_demangling=no
      ;;
   yes)
      AC_MSG_ERROR([cannot find any library for the backend supported by --enable-demangling])
      ;;
   *)
      AC_MSG_ERROR([bad value '$enable_demangling' for --enable-demangling])
      ;;
esac
```